### ISSUE TITLE
Implement exporting only a subset of the current file as an animation

### ISSUE THREAD

**Issue by TiborGY** [OpenChemistry/avogadrolibs#995]
Sometimes files can have a **lot** of geometries:
![image](https://user-images.githubusercontent.com/29861150/177046011-6fdb34ff-1645-46a4-9ba4-f1b4f0ad737a.png)

It would be nice if Avogadro asked the user how many frames of the file should be exported to the MP4/GIF/AVI when the user clicks "Record movie...", especially as it takes quite a long time to export 200k+ frames.

Should be doable with a new dialog popup, but I am no Qt expert. Instead of the number of frames, the first & last frame could also be asked, or a duration, or some combination. But as a basic feature only asking the number of frames wanted should be sufficient.

It should be validated of course that it does not exceed the number of remaining frames, and the user should be warned if it does.

**Comment by ghutchis** [comment]
Good points. It might also be useful for files with many coordinates to not load all at once (e.g., large trajectories of proteins).

Do you run into slowdowns loading the file?

**Comment by TiborGY** [comment]
Not really, it does take a bit of time to load but its not too bad.
But then again, I have a Ryzen 3600 and 32 GiB of RAM running faster than JEDEC speeds.

What **is** slow, very slow in fact is the "Record movie" export process. It is also one where A2 UI simply freezes (i.e. "Not responding" under Windows) while it continues working on the export at glacial speeds.

**Comment by ghutchis** [comment]
Yes, I think the "export movie" happens in the main thread. That's definitely something to fix.

**Comment by ghutchis** [comment]
I had an idea about this last night. Right now, there's a line with "Frame 1 of 223862".

How about making that line with three boxes:
Frame: [start frame]   [ current frame ]  [ end frame ] "out of 223862"

Then you can set exactly the subset you want (and the UI itself would only show that part of the animation).

**Comment by ghutchis** [comment]
<img width="265" alt="Image" src="https://github.com/user-attachments/assets/896e06a1-dfdb-4e2e-91b0-8e954064cb8d" />
